### PR TITLE
Fix GPUXToSpirv Pass to iterate for all gpuModules

### PR DESCRIPTION
**Pull request by nbpatel**
Please review these guidelines to help with the review process:
- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, a reproducer, or a reference to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [x] If this PR is a work in progress, are you filing the PR as a draft?
- [x] Have you organized your commits logically and ensured each can be built by itself?

### CODE REVIEW

File: lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp [hunk 1]
```diff
@@ -64,43 +64,47 @@ void GPUXToSPIRVPass::runOnOperation() {
...
   });
 
-  // Map MemRef memory space to SPIR-V storage class first if requested.
-  if (mapMemorySpace) {
-    std::unique_ptr<mlir::ConversionTarget> target =
-        mlir::spirv::getMemorySpaceToStorageClassTarget(*context);
-    mlir::spirv::MemorySpaceToStorageClassMap memorySpaceMap =
-        mlir::spirv::mapMemorySpaceToVulkanStorageClass;
-    mlir::spirv::MemorySpaceToStorageClassConverter converter(memorySpaceMap);
+  for (auto gpuModule : gpuModules) {
```
Comment (drprajap) [inline lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp:67]: so this means upstream is not supporting iterating over all gpu modules?

Comment (nbpatel) [reply lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp:67]: upstream does it as well ..here https://mlir.llvm.org/doxygen/GPUToSPIRVPass_8cpp_source.html ..line 68

**Review by silee2 — APPROVED** [review]
LGTM

File: lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp (hunk 1 above)
Comment (drprajap) [reply lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp:67]: Yes, I see it now, when I added this pass, upstream did not have this covered. Looks good.

**Review by drprajap — APPROVED** [review]

File: lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp (hunk 1 above)
Comment (Hardcode84) [reply lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp:67]: I did this change both in `main` and in upstream, to support attaching different TargetEnv's to different gpu modules https://reviews.llvm.org/D135907

**Pull request merged**